src/core/Badge/Badge.tsx

```diff
@@ -7,13 +7,48 @@ import cx from 'classnames';
 import { CommonProps } from '../utils/props';
 import { useTheme } from '../utils/hooks/useTheme';
 import '@itwin/itwinui-css/css/badge.css';
+import { getWindow, isSoftBackground, SoftBackgrounds } from '../utils/common';
+
+/**
+ * Helper function that returns one of the preset badge color values.
+ */
+const getBadgeColorValue = (color: BadgeProps['backgroundColor']) => {
+  if (!color) {
+    return '';
+  }
+
+  switch (color) {
+    case 'primary':
+      return '#A5D7F5';
+    case 'positive':
+      return '#C3E1AF';
+    case 'negative':
+      return '#EFA9A9';
+    case 'warning':
+      return '#F9D7AB';
+    default:
+      if (isSoftBackground(color)) {
+        return SoftBackgrounds[color];
+      }
+      return color;
+  }
+};
 
 export type BadgeProps = {
   /**
    * Background color of the badge.
-   * Recommended to use one of the data visualization colors: `$iui-color-dataviz-*`.
+   *
+   * Recommended to use one of the preset colors for statuses and soft backgrounds.
+   *
+   * If not specified, a default neutral background will be used.
    */
-  backgroundColor: string;
+  backgroundColor?:
+    | 'primary'
+    | 'positive'
+    | 'negative'
+    | 'warning'
+    | keyof typeof SoftBackgrounds
+    | (string & {}); // eslint-disable-line @typescript-eslint/ban-types -- see https://github.com/Microsoft/TypeScript/issues/29729
```

Comment:
Maybe add a comment that it allows predefined values and custom strings, also intellisense is working.

Reply:
Added